test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -0,0 +1,533 @@
+        }
+
+        [Theory]
+        [MemberData(nameof(ListBoxDataSet))]
+        public void ListBoxFor_WithNullModel_GeneratesExpectedValue(
+            IEnumerable<SelectListItem> selectList,
+            string expectedHtml,
+            string ignoredHtml1,
+            string ignoredHtml2)
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper<ModelContainingList>(model: null);
+            var savedSelected = selectList.ToDictionary(item => item.Text, item => item.Selected);
+
+            // Act
+            var html = helper.ListBoxFor(value => value.Property1, selectList, htmlAttributes: null);
+
+            // Assert
+            Assert.Equal(expectedHtml, html.ToString());
+            Assert.Equal(savedSelected, selectList.ToDictionary(item => item.Text, item => item.Selected));
+        }
+
+        [Theory]
+        [MemberData(nameof(ListBoxDataSet))]
+        public void ListBoxFor_WithModelValue_GeneratesExpectedValue(
+            IEnumerable<SelectListItem> selectList,
+            string ignoredHtml1,
+            string expectedHtml,
+            string ignoredHtml2)
+        {
+            // Arrange
+            var model = new ModelContainingList { Property1 = { "2" } };
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(model);
+            var savedSelected = selectList.ToDictionary(item => item.Text, item => item.Selected);
+
+            // Act
+            var html = helper.ListBoxFor(value => value.Property1, selectList, htmlAttributes: null);
+
+            // Assert
+            Assert.Equal(expectedHtml, html.ToString());
+            Assert.Equal(savedSelected, selectList.ToDictionary(item => item.Text, item => item.Selected));
+        }
+
+        [Theory]
+        [MemberData(nameof(ListBoxDataSet))]
+        public void ListBoxFor_WithMultipleModelValues_GeneratesExpectedValue(
+            IEnumerable<SelectListItem> selectList,
+            string ignoredHtml1,
+            string ignoredHtml2,
+            string expectedHtml)
+        {
+            // Arrange
+            var model = new ModelContainingList { Property1 = { "1", "3" } };
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(model);
+            var savedSelected = selectList.ToDictionary(item => item.Text, item => item.Selected);
+
+            // Act
+            var html = helper.ListBoxFor(value => value.Property1, selectList, htmlAttributes: null);
+
+            // Assert
+            Assert.Equal(expectedHtml, html.ToString());
+            Assert.Equal(savedSelected, selectList.ToDictionary(item => item.Text, item => item.Selected));
+        }
+
+        private class ModelContainingList
+        {
+            public ModelContainingList()
+            {
+                Property1 = new List<string>();
+            }
+
+            public List<string> Property1 { get; }
```

Comment:
`public List<string> Property1 { get; } = new List<string>();` and drop the ctor.